test/Errors.hs

```diff
@@ -23,15 +23,11 @@ import           Data.Time (Day, UTCTime)
 import           Common
 import           Database.SQLite.Simple.Types (Null)
 
--- The "length (show e) `seq` .." trickery below is to force evaluate
--- the contents of error messages.  Another option would be to log
--- them (would be useful), but I don't know if HUnit has any logging
--- mechanisms.  Just printing them as is will look like the tests are
--- hitting errors and would be confusing.
-assertResultErrorCaught :: IO a -> Assertion
-assertResultErrorCaught action = do
-  catch (action >> return False) (\(e :: ResultError) -> length (show e) `seq` return True) >>=
-    assertBool "assertResultError exc"
+assertResultErrorThrown :: IO a -> ResultError -> Assertion
+assertResultErrorThrown action expectedError =
```

Comment:
This is I think better way to test exception throwing behavior - actually compare the expected exception with what's being thrown.